config/db_router.py

```diff
@@ -0,0 +1,63 @@
+from django.conf import settings
+from django.db import DEFAULT_DB_ALIAS
+
+
+class SecondaryDatabaseRouter:
+    """
+    A router to direct migration operations for specific models to the secondary database.
+    """
+
+    def db_for_read(self, model, **hints):
+        # Prevent reads from the secondary database
+        return DEFAULT_DB_ALIAS
+
+    def db_for_write(self, model, **hints):
+        # Prevent writes to the secondary database
+        return DEFAULT_DB_ALIAS
+
+    def allow_relation(self, obj1, obj2, **hints):
+        # Allow relations only in the default database
+        return None
+
+    def allow_migrate(self, db, app_label, model_name=None, **hints):
+        if db == DEFAULT_DB_ALIAS:
+            return True
+        elif db == settings.SECONDARY_DB_ALIAS:
+            # Data migrations using RunPython don't need to be
```

Comment:
Note we do need all the table schemas (not just the superset tables) even though we enable replication for just  [REPLICATION_ALLOWED_MODELS](https://github.com/dimagi/commcare-connect/pull/481/files#diff-f040749dbea6f228246cc9aead49904004997f8f6212981561e1d4d27ccd91a2R18). This is because django migrations can't work without all schemas in place.

Another notable thing here is that data migrations targetted using `RunPython` module won't be propagated to secondary database as data is replicated at database level.